src/components/HorizontalStepper.tsx

```diff
@@ -0,0 +1,228 @@
+import { FC, ReactNode } from 'react';
+import {
+  Stepper,
+  StepperProps,
+  Step,
+  StepProps,
+  StepLabel,
+  StepLabelProps,
+  Divider,
+  Button,
+  Stack,
+  Box,
+  Typography,
+  Container,
+  useTheme,
+  alpha,
+} from '@mui/material';
+import { Check } from '@mui/icons-material';
+
+export type StepType = {
+  id: string;
+  label: string;
+  content: (info: { id: string, activeStep: number }) => ReactNode;
+  stepProps?: StepProps;
+  stepLabelProps?: StepLabelProps;
+};
+
+export type HorizontalStepperProps = StepperProps & {
+  steps: StepType[];
+  disabled?: boolean;
+  onNext?: () => void;
+  disabledNext?: boolean;
+  onBack?: () => void;
+  disabledBack?: boolean;
+  onFinish?: () => void;
+  disabledFinish?: boolean;
+  getStepStateLabel?: (step: number) => string | null;
+  getBackLabel?: (step: StepType) => string | null;
+  getNextLabel?: (step: StepType) => string | null;
+  finishLabel?: string;
+};
+
+export const HorizontalStepper: FC<HorizontalStepperProps> = (props) => {
+  const {
+    steps,
+    activeStep = 0,
+    disabled = false,
+    onNext = () => null,
+    disabledNext = false,
+    onBack = () => null,
+    disabledBack = false,
+    onFinish = () => null,
+    disabledFinish = false,
+    getStepStateLabel = () => null,
+    getBackLabel = () => null,
+    getNextLabel = () => null,
+    finishLabel = null,
+    ...stepperProps
+  } = props;
```

Comment:
```suggestion
  const {
    steps,
    activeStep = 0,
    disabled = false,
    nextButton?: {
      onNext = () => null,
      disabledNext = false,
      getNextLabel = () => null,
    },
    backButton?: {
      onBack = () => null,
      disabledBack = false,
      getBackLabel = () => null,
    },
    finishButton?: {
      onFinish = () => null,
      disabledFinish = false,
      finishLabel = null,
    },
    getStepStateLabel = () => null,
    ...stepperProps
  } = props;
```

Qué te parece juntar las props de cada boton en un objeto? Para tener menos props digo